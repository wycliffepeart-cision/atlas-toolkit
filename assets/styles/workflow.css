@layer atlas {
	atlas-workflow {
		display: block;
		width: 100%;
	}
}
