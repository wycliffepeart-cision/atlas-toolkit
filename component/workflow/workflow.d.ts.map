{"version":3,"file":"workflow.d.ts","sourceRoot":"","sources":["../../../src/component/workflow/workflow.ts"],"names":[],"mappings":"AAGA,qBACa,QAAS,SAAQ,WAAW;CAAG"}