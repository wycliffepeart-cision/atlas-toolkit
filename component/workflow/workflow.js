import { _ as __decorate } from '../../tslib.es6-2e0713a3.js';
import { Template } from '../../decorators/template.js';

var WorkflowTemplate = "<template>\n  <slot part=\"steps\"></slot>\n</template>\n";

let Workflow = class Workflow extends HTMLElement {
};
Workflow = __decorate([
    Template(WorkflowTemplate)
], Workflow);
if (!customElements.get('atlas-workflow'))
    customElements.define('atlas-workflow', Workflow);

export { Workflow };
