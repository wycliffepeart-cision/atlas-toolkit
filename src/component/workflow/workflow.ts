import { Template } from '../../decorators/template';
import WorkflowTemplate from './workflow.html';

@Template(WorkflowTemplate)
export class Workflow extends HTMLElement {}

if (!customElements.get('atlas-workflow')) customElements.define('atlas-workflow', Workflow);

