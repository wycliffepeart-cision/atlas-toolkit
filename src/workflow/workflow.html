<template>
	<slot part="steps"></slot>
</template>
