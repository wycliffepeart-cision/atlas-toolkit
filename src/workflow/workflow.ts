import { Template } from '../decorators/template'
import WorkflowTemplate from './workflow.html'

@Template(WorkflowTemplate)
export class Workflow extends HTMLElement {}
