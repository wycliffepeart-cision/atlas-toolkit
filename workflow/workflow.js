import { _ as __decorate } from '../tslib.es6-806de115.js';
import { Template } from '../decorators/template.js';

var WorkflowTemplate = "<template>\n  <slot part=\"steps\"></slot>\n</template>\n";

let Workflow = class Workflow extends HTMLElement {
};
Workflow = __decorate([
    Template(WorkflowTemplate)
], Workflow);

export { Workflow };
