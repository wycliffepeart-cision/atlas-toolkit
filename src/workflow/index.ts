import { Workflow } from './workflow';

if (!customElements.get('atlas-workflow')) customElements.define('atlas-workflow', Workflow);
