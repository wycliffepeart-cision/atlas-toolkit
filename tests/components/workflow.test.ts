import '../../src/component/workflow/workflow';

const WORKFLOW_TAG = 'atlas-workflow';

afterEach(() => {
  const elem = document.querySelector('body > *:first-child');
  elem?.remove();
});

describe('Workflow', () => {
  it('is correctly appended in the DOM', () => {
    const elem = document.createElement(WORKFLOW_TAG);
    document.body.appendChild(elem);

    (expect(elem) as any).toBeInTheDocument();
  });

  it('is matches snapshot', () => {
    const elem = document.createElement(WORKFLOW_TAG);
    document.body.appendChild(elem);

    (expect(elem.shadowRoot.innerHTML) as any).toMatchSnapshot();
  });
});
