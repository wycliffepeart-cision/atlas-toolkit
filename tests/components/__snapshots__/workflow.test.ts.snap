// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Workflow is matches snapshot 1`] = `
<slot part="steps">
</slot>
`;
